organization/_posts/2018/2018-12-05-softwareforum.md

```diff
@@ -0,0 +1,98 @@
+---
+title: "Spark based analysis for LSST and HEP, December 5, 2018"
+layout: meetings
+---
+
+# {{page.title}}
+
+[Meeting Agenda and Slides](https://indico.cern.ch/event/754811/)
+
+Introduction
+------------
+-   Reminder about the [*HSF/OSG/WLCG
+    Workshop*](https://indico.cern.ch/event/759388/), 18-22 March 2019
+    at JLab.
+    -   We hope to open registration this week.
+-   HSF intends to submit a short (1 page) paper to the European
+    Strategy for Particle Physics Review Process.
+    -   Will be circulated tomorrow for 1 week.
+-   New working groups are up and running.
+    -   Will have a report from the convenors in next week’s HSF
+        Coordination meeting.
+
+Analyzing astronomical data with Apache Spark
+---------------------------------------------
+-   Simulation data can be N-body simulations of the evolution of the
+    universe.
+-   An object for LSST is anything that can be “labeled” on the sky:
+    -   Like a data frame row - with 100 or 1000 columns (position,
+        redshift, spectra, …)
+    -   Data is flatter than in particle physics (usually completely
+        flat in the final catalogue).
+-   FITS data does not fit “naturally” into Spark, but have written an adaptor.
+-   Performance test was for \~small data elements (5 doubles); total
+    size 100GB.
+-   The schema for the dataset is fixed for each case; but see issues when
+    many input files are used (causing the metadata bottleneck).
+-   3D spatial data not well supported in Spark ecosystem.
+    -   Most packages work with 2D data.
+-   Re-partitioning in space is only one option - in general want to be
+    able to repartition on arbitrary metrics:
+    -   Need to build partition and then shuffle the data (2 phases).
+    -   UDF = User Defined Function - this is difficult to optimise as
+        it’s a black box for Spark.
+-   For C/C++ interfaces how to serialise?
+    -   Contact Christan Arnault for this.
+-   Training - how can people run the tutorials?
+    -   Targeting LSST people and they mostly have NERSC accounts where
+        clusters are available. LAL cluster also available (which may
+        expand if it proves popular).
+-   For partitioning, isn’t Spark specifically designed for this kind of
+    task?
+    -   For a lot of cases Spark does do a good job, but if the data has
+        structures that were not known at data loading time then the
+        map/reduce workflow performs poorly when the output is as
+        large as the input.
+    -   Q. Problem with the memory usage? Is the algorithm optimal?
+        - Default Spark partitioning is not doing so well, hence
+          the O(N^2) scaling seen.
+
+Distributed data analysis with ROOT RDataFrame and Spark
+--------------------------------------------------------
+-   Working with ROOT to establish working model for analysis for FCC
```

Comment:
I'm a bit confused here, I mentioned FCC to present myself however this project is not related to FCC at the moment.